spec/lib/payment_processor/clients/braintree/webhook_handler_spec.rb

```diff
@@ -1,118 +0,0 @@
-                it 'pushes the transaction to be queued' do
-                  expect(ChampaignQueue).to have_received(:push).
-                    with({type: "subscription-payment", params: { recurring_id: "1234" }})
-                end
-              end
-
-              context 'with no existing transactions' do
-                let(:transactions) { [] }
-
-                it 'pushes the transaction to be queued' do
-                  expect(ChampaignQueue).to have_received(:push)
-                end
-              end
-
-              it 'records to Payment.write_transaction' do
-                expect(Payment).to have_received(:write_transaction).with(
-                  notification, action.page_id, action.member_id, nil, false
-                )
-              end
-
-              it 'does not log anything' do
-                expect(Rails.logger).not_to have_received(:info)
-              end
-            end
-
-            describe 'when Action is not found' do
-              before :each do
-                allow(Payment::BraintreeSubscription).to receive(:find_by).and_return(nil)
-                WebhookHandler.handle(notification)
-              end
-
-              it 'does not write to Payment.write_transaction' do
-                expect(Payment).not_to have_received(:write_transaction)
-              end
-
-              it 'does not push to ChampaignQueue' do
-                expect(ChampaignQueue).not_to have_received(:push)
-              end
-
-              it 'logs the failed handling' do
-                expect(Rails.logger).to have_received(:info).with("Failed to handle Braintree::WebhookNotification for subscription_id 's09870'")
-              end
-            end
-          end
-
-          describe 'with subscription cancelation' do
-
-            let(:notification) do
-              instance_double('Braintree::WebhookNotification', kind: 'subscription_canceled',
-                subscription: instance_double('Braintree::Subscription', id: 's09870')
-              )
-            end
-
-            before :each do
-              allow(Payment::BraintreeSubscription).to receive(:find_by)
-              WebhookHandler.handle(notification)
-            end
-
-            it 'does not look up the action' do
-              expect(Payment::BraintreeSubscription).not_to have_received(:find_by)
-            end
-
-            it 'does not write to Payment.write_transaction' do
-              expect(Payment).not_to have_received(:write_transaction)
-            end
-
-            it 'does not push to ChampaignQueue' do
-              expect(ChampaignQueue).not_to have_received(:push)
-            end
-
-            it 'logs the failed handling' do
-              expect(Rails.logger).to have_received(:info).with("Unsupported Braintree::WebhookNotification received of type 'subscription_canceled'")
-            end
-          end
-        end
-      end
-    end
-  end
-end
```

Comment:
This isn't the best spec file. Decided to delete it since behaviour is already covered `spec/requests/api/braintree_webhook_spec.rb` (without all of the stubs and mocks).